frontend/ui/widget/filechooser.lua

```diff
@@ -62,7 +62,11 @@ function FileChooser:init()
         if ok then
             for f in iter, dir_obj do
                 if count_only then
-                    if self.show_hidden or not util.stringStartsWith(f, ".") then
+                    if
+                        not self.show_hidden and not util.stringStartsWith(f, ".")
+                    or
+                        (self.show_hidden and f ~= "." and f ~= ".." and self.dir_filter(f) and not util.stringStartsWith(f, "._"))
+                    then
```

Comment:
Not fond of that syntax :) (`if or then` on the same indent level - `if` with nothing following it.
I'd rather have multiple if / elseif with the action (small table.insert) duplicated.
If you prefer keeping a single if, I'd then prefer to have parens around the first alternative `if (not and not) or (and and and not)`, make it easier to understand and not worry about logical operator precedence (and > or).

And in the first alternative, won't you be including the `._` Mac ressources?

Reply:
Also starts with `.`, doesn't it? ;-)

Reply:
Oh, right :| :)